kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java

```diff
@@ -347,6 +354,10 @@ public TypeDeserializer buildTypeDeserializer(DeserializationConfig config, Java
             typer2 = typer2.init(JsonTypeInfo.Id.CLASS, null);
             typer2 = typer2.inclusion(JsonTypeInfo.As.WRAPPER_OBJECT);
             deserializeObjectMapper.setDefaultTyping(typer2);
+            // this avoids the use of default typing
+            if(!defaultTypingEnabled) {
+                deserializeObjectMapper.deactivateDefaultTyping();
```

Comment:
if we disable default typing here, wouldn't it make sense to wrap lines 327-356 in a `if(defaultTypingEnabled)` clause?

Reply:
done